src/crimson/common/smp_helpers.h

```diff
@@ -15,6 +15,7 @@
 #include "common/likely.h"
 #include "crimson/common/errorator.h"
 #include "crimson/common/utility.h"
+#include "crimson/common/coroutine.h"
```

Comment:
the title of the commit reads:

> crimson/common/smp_helpers: missing include

please use imperative mood, and would be better to include the compiling error message addressed by this change to help reviewers to understand your fix.

see also https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst#commit-title

Reply:
https://github.com/ceph/ceph/pull/63523#discussion_r2168238355

This PR causes gcc 15 builds failed.

Reply:
> [#63523 (comment)](https://github.com/ceph/ceph/pull/63523#discussion_r2168238355)
> 
> This PR causes gcc 15 builds failed.

@zhscn thanks for pointing this out. but it does not address my questions / suggestion above.

Reply:
Yeah, I mean this part.

> looks like part of a bug report.

The `seastar_task_ancestor` class in `crimson/common/coroutine.h` inherits from `seastar::task`. Neither class defines a virtual destructor, which triggers the `-Wnon-virtual-dtor` warning. Combined with `-Werror`, this causes the build to fail.

Reply:
@zhscn i understand the reason why we have this change. but we also need to explain the reason why we need to have it in commit message for posterity. did you get a chance to read my comment at https://github.com/ceph/ceph/pull/64203#discussion_r2173723975 ?